### PR TITLE
[FEATURE] Postgres 15 support

### PR DESCRIPTION
**Pull request by raphaelauv**
Postgres 15 have been out for 2 weeks , let's add support

### CODE REVIEW

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
I think we could also add some "postgres-excludes" (in selective_checks) - this way we are not growing the whole matrix of text too much - we do not really need to run absolute complete matrix of python-versios and database versions - we are ok to exclude arbitrary some python versions for some databases. With another postgres version added we add many more python jobs so I think adding "3.8" to postgres-exclude?

File: dev/breeze/SELECTIVE_CHECKS.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ The selective check outputs available are described below:
 | needs-helm-tests                   | Whether Helm tests are needed to run ("true"/"false")                                                  | true                                                          |
 | needs-javascript-scans             | Whether javascript CodeQL scans should be run ("true"/"false")                                         | true                                                          |
 | needs-python-scans                 | Whether Python CodeQL scans should be run ("true"/"false")                                             | true                                                          |
-| postgres-exclude                   | Which versions of Postgres to exclude for tests as JSON array                                          | []                                                            |
+| postgres-exclude                   | Which versions of Postgres to exclude for tests as JSON array                                          | ['3.8']                                                       |
```
Comment (jedcunningham) [inline dev/breeze/SELECTIVE_CHECKS.md]: This also needs to be done here to actually work:
https://github.com/apache/airflow/blob/8f5e100f30764e7b1818a336feaa8bb390cbb327/dev/breeze/src/airflow_breeze/utils/selective_checks.py#L356

Comment (potiuk) [reply dev/breeze/SELECTIVE_CHECKS.md]: yep - what Jed wrote is what's needed.

**Comment by jedcunningham** [comment]
Now that we can drop 10 (see #27594), we can probably just add 15?

**Comment by potiuk** [comment]
And even approved by you @jedcunningham :wink:

**Comment by jedcunningham** [comment]
Right, I just mean we can just run 15 against all python versions instead of excluding 3.8 to reduce the matrix size (which is what this was waiting on).

**Comment by potiuk** [comment]
I rebased it - it looked like it was needed.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
> Right, I just mean we can just run 15 against all python versions instead of excluding 3.8 to reduce the matrix size (which is what this was waiting on).

@jedcunningham : I think this one should handle it in a more "holistic" way https://github.com/apache/airflow/pull/27754

**Comment by potiuk** [comment]
Merged this one. With https://github.com/apache/airflow/pull/27754 it should run WAY faster in main - only adding one more build to the current matrix.

**Pull request merged**